src/main/java/org/embulk/input/zendesk/utils/ZendeskValidatorUtils.java

```diff
@@ -0,0 +1,136 @@
+                            task.getAuthenticationMethod().name()));
+                }
+                break;
+            case BASIC:
+                if (!task.getUsername().isPresent() || !task.getPassword().isPresent()) {
+                    throw new ConfigException(String.format("username and password are required for authentication method '%s'",
+                            task.getAuthenticationMethod().name()));
+                }
+                break;
+            default:
+                throw new ConfigException("Unknown authentication method");
+        }
+
+        // Validate credentials by sending one request to users.json. It Should always have at least one user
+        try {
+            zendeskSupportAPIService.getData(String.format("%s%s/users.json?per_page=1", task.getLoginUrl(),
+                    ZendeskConstants.Url.API), 0, false);
+        }
+        catch (final HttpResponseException ex) {
+            if (ex.getResponse().getStatus() == 401) {
+                throw new ConfigException("Invalid credential. Error 401: can't authenticate");
+            }
+        }
+    }
+
+    private static void validateAppMarketPlace(final boolean isAppMarketIntegrationNamePresent,
+                                        final boolean isAppMarketAppIdPresent,
+                                        final boolean isAppMarketOrgIdPresent)
+    {
+        final boolean isAllAvailable =
+                isAppMarketIntegrationNamePresent && isAppMarketAppIdPresent && isAppMarketOrgIdPresent;
+        final boolean isAllUnAvailable =
+                !isAppMarketIntegrationNamePresent && !isAppMarketAppIdPresent && !isAppMarketOrgIdPresent;
+        // All or nothing needed
+        if (!(isAllAvailable || isAllUnAvailable)) {
+            throw new ConfigException("All of app_marketplace_integration_name, app_marketplace_org_id, " +
+                    "app_marketplace_app_id " +
+                    "are required to fill out for Apps Marketplace API header");
+        }
+    }
+
+    private static void validateIncremental(final ZendeskInputPlugin.PluginTask task)
+    {
+        // auto run with non incremental if target doesn't support incremental
+        if (!ZendeskUtils.isSupportIncremental(task.getTarget())) {
+            task.setIncremental(false);
+        }
+
+        if (task.getIncremental()) {
+            if (!task.getDedup()) {
+                logger.warn("You've selected to skip de-duplicating records, result may contain duplicated data");
+            }
+        }
+        else {
+            if (task.getStartTime().isPresent()) {
+                task.setStartTime(Optional.of(ZendeskConstants.Misc.DEFAULT_START_TIME));
+            }
+
+            if (ZendeskUtils.isSupportIncremental(task.getTarget())) {
+                task.setIncremental(true);
+            }
+        }
+
+        if (task.getIncremental()) {
+            if (!ZendeskDateUtils.isSupportedTimeFormat(task.getStartTime().get(),
```

Comment:
Should move this move to above (`if` incremental) block? Besides, if config has no `start_time` and no `incremental` (ie. default to `true`), it will throw error:
```
java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at org.embulk.input.zendesk.utils.ZendeskValidatorUtils.validateIncremental(ZendeskValidatorUtils.java:128)
	at org.embulk.input.zendesk.utils.ZendeskValidatorUtils.validateInputTask(ZendeskValidatorUtils.java:31)
	at org.embulk.input.zendesk.ZendeskInputPlugin.guess(ZendeskInputPlugin.java:195)
```
Best practice is always check for `#isPresent()` before `#get()`